File: src/main/java/net/azzy/pulseflux/registry/ParticleRegistry.java
package net.azzy.pulseflux.registry;

import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.fabricmc.fabric.api.particle.v1.FabricParticleTypes;
import net.minecraft.particle.DefaultParticleType;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

import static net.azzy.pulseflux.PulseFlux.MOD_ID;

public class ParticleRegistry {


    @Environment(EnvType.CLIENT)
    public static void initClient() {
    }

    public static DefaultParticleType register(String name, boolean alwaysShow) {
        return Registry.register(Registry.PARTICLE_TYPE, new Identifier(MOD_ID, name), FabricParticleTypes.simple(alwaysShow));
    }
}

File: src/main/java/net/azzy/pulseflux/registry/ParticleRegistry.kt
package net.azzy.pulseflux.registry

import net.azzy.pulseflux.PulseFlux
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.fabricmc.fabric.api.particle.v1.FabricParticleTypes
import net.minecraft.particle.DefaultParticleType
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

object ParticleRegistry {
    @JvmStatic
    @Environment(EnvType.CLIENT)
    fun initClient() {
    }

    fun register(name: String?, alwaysShow: Boolean): DefaultParticleType {
        return Registry.register(Registry.PARTICLE_TYPE, Identifier(PulseFlux.MOD_ID, name), FabricParticleTypes.simple(alwaysShow))
    }
}